clients/roscpp/src/libros/io.cpp

```diff
@@ -213,18 +289,33 @@ int poll_sockets(socket_pollfd *fds, nfds_t nfds, int timeout) {
 		} else {
 				fds[i].revents = POLLNVAL;
 		}
+                ofds->push_back(fds[i]);
 	}
-	return rc;
+	return ofds;
 #else
-	// use an existing poll implementation
-	int result = poll(fds, nfds, timeout);
-	if ( result < 0 ) {
-		// EINTR means that we got interrupted by a signal, and is not an error
-		if(errno == EINTR) {
-			result = 0;
-		}
-	}
-	return result;
+        UNUSED(nfds);
+        UNUSED(fds);
+        struct epoll_event ev[nfds];
+        boost::shared_ptr<std::vector<socket_pollfd> > ofds;
+
+        int fd_cnt = epoll_wait(epfd, ev, nfds, timeout);
+
+        if (fd_cnt < 0)
+        {
+          ROS_ERROR("Error in epoll_wait! %s", strerror(errno));
```

Comment:
The answer is "no". don't know when this would happen. And the best I can do now is retry next time hoping the failure will go away. 

I am suspecting that may be the computer going into sleep mode might cause the syscall to fail on return. But have not tested it thoroughly yet. There are some flags to control that part which I want to play with once the overall structure is solidified.